shared-libs/cht-datasource/src/local/libs/doc.ts

```diff
@@ -0,0 +1,12 @@
+import { Nullable } from '../../libs/core';
+import { Doc, isDoc } from '../../libs/doc';
+
+export const getDocById = (db: PouchDB.Database<Doc>) => async (uuid: string): Promise<Nullable<Doc>> => db
```

Comment:
I have gone back and forth with this a couple times.  My main concern is that we don't accidentally change the return types of _functions exported and used outside of cht-datasource_.  That could be a bit tricky to detect and lead to some interesting debugging.  

Currently, I am [using the `explicit-module-boundary-types` eslint rule](https://github.com/medic/cht-core/blob/5196de4b1cb132c90cb996ab2d705d45907e58b0/shared-libs/cht-datasource/.eslintrc.js#L18) to require return types on _exported_ functions (applies to all exported functions, not just ones that end up getting exported from `index.ts`, but it was the best I could find).  The [rule docs](https://typescript-eslint.io/rules/explicit-module-boundary-types/) do note that return types on module boundaries "can also improve TypeScript type checking performance on larger codebases". Not something I am too concerned about at this point, but is a nice feature...

 At the end of the day, I feel like our unit tests should be a sufficient backstop to prevent us from accidentally breaking things with regards to the data returned from public functions (and we def should not be heavily relying on the _types_ to save us here... :sweat_smile: ). So, I am happy to go either way. Let me know if the `explicit-module-boundary-types` seems valuable given the above rational, otherwise I can just remove it!  (Another interesting datapoint is that `explicit-module-boundary-types` was not included in the `strict-type-checked` or the `stylistic-type-checked` eslint rulesets which tells me there is not broad consensus that it is a "best practice".)

Reply:
So I went back to [this video](https://www.totaltypescript.com/tips/dont-use-return-types-unless) (the author made a great typescript course!) and he summarized it at the end:
> To sum up, in application code, don't by default use return types. Use them if your function has multiple branches. Use them if you're writing a library, definitely. Use them if you have any weird niche performance concerns to do with really big objects being passed and inferred through your code.

In the video he made an example of why exported library functions should be typed and that made a lot of sense! Plus, by explicitly typing these functions, we get an additional safety layer for javascript projects that import the compiled code without any types. Let's keep it that way :+1:

Reply:
:+1: I am going to start a list of "do's" and "don'ts" for TS code in medic repos where I will capture some of these design thoughts.  Maybe can put it out in the Dev Handbook on the docs site...